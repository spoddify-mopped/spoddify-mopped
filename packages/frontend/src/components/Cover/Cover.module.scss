div.cover {
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 10rem;
  padding: 1rem;
  margin: 1rem;
  cursor: pointer;
  border-radius: 5px;
  background-color: rgba($color: #ffffff, $alpha: 0.05);

  &:hover {
    background-color: rgba($color: #ffffff, $alpha: 0.15);
  }

  img {
    width: 10rem;
    height: 10rem;
    margin-bottom: 1rem;
  }

  span.title {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    font-weight: 500;
  }

  span.subtitle {
    opacity: 0.5;
    font-size: 0.9rem;

    &.clickable {
      &:hover {
        text-decoration: underline;
      }
    }
  }
}
