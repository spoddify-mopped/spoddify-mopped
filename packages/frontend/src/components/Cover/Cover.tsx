import React, { useRef } from 'react';

import styles from './Cover.module.scss';

type CoverItem = {
  clickableSubtitle?: boolean;
  id: string;
  name: string;
  image?: string;
  subtitle?: React.ReactElement | React.ReactElement[];
  onClick?: () => void;
};

type Props = {
  item: CoverItem;
  className?: string;
};

const Cover = ({ className, item }: Props): React.ReactElement => {
  const subtitle = useRef<HTMLSpanElement>(null);

  return (
    <div
      key={`cover_${item.id}_${item.name}`}
      className={`${styles.cover} ${className || ''}`}
      onClick={(evt) => {
        if (item.clickableSubtitle && subtitle.current) {
          const children = subtitle.current.children;
          for (let i = 0; i < children.length; i++) {
            if (evt.target === children[i]) {
              return;
            }
          }
        }

        item.onClick && item.onClick();
      }}
    >
      <img src={item.image} alt="Thumb" />
      <span title={item.name} className={styles.title}>
        {item.name}
      </span>
      <span ref={subtitle} className={`${styles.subtitle}`}>
        {item.subtitle}
      </span>
    </div>
  );
};

export default Cover;
