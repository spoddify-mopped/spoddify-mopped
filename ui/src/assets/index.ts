import { ReactComponent as AddIcon } from './svgs/add.svg';
import CoverReplacement from './images/cover_replacement.png';
import { ReactComponent as FullscreenIcon } from './svgs/fullscreen.svg';
import { ReactComponent as NextIcon } from './svgs/next.svg';
import { ReactComponent as PauseIcon } from './svgs/pause.svg';
import { ReactComponent as PlayCircleIcon } from './svgs/play-circle.svg';
import { ReactComponent as PlayIcon } from './svgs/play.svg';
import { ReactComponent as PlaylistIcon } from './svgs/playlist.svg';
import { ReactComponent as PreviousIcon } from './svgs/previous.svg';
import { ReactComponent as SearchIcon } from './svgs/search.svg';
import { ReactComponent as SettingsIcon } from './svgs/settings.svg';
//import { ReactComponent as SpoddifyMoppedLogo } from './svgs/logo.svg';
import { ReactComponent as SpoddifyMoppedLogo } from './svgs/spoddify-mopped-logo.svg';
import { ReactComponent as VolumeIcon } from './svgs/volume.svg';

export {
  AddIcon,
  CoverReplacement,
  FullscreenIcon,
  NextIcon,
  PauseIcon,
  PlayCircleIcon,
  PlayIcon,
  PlaylistIcon,
  PreviousIcon,
  SearchIcon,
  SettingsIcon,
  SpoddifyMoppedLogo,
  VolumeIcon,
};
