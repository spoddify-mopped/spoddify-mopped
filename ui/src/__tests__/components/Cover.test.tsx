import Cover from '../../components/Cover/Cover';
import { CoverReplacement } from '../../assets';
import React from 'react';
import renderer from 'react-test-renderer';

it('Cover renders correctly with required item properties', () => {
  const tree = renderer
    .create(
      <Cover
        item={{
          id: 'some id',
          name: 'some name',
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Cover renders correctly with custom class name', () => {
  const tree = renderer
    .create(
      <Cover
        item={{
          id: 'some id',
          name: 'some name',
        }}
        className="some class"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Cover renders correctly with image', () => {
  const tree = renderer
    .create(
      <Cover
        item={{
          id: 'some id',
          image: 'some image url',
          name: 'some name',
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Cover renders correctly with image and onClick', () => {
  const tree = renderer
    .create(
      <Cover
        item={{
          id: 'some id',
          image: CoverReplacement,
          name: 'some name',
          onClick: () => console.log('Test'),
        }}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('Cover renders correctly with subtitle', () => {
  const tree = renderer
    .create(
      <Cover
        item={{
          id: 'some id',
          image: CoverReplacement,
          name: 'some name',
          onClick: () => console.log('Test'),
          subtitle: <span>'some subttle'</span>,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Cover renders correctly with clickable subtitle', () => {
  const tree = renderer
    .create(
      <Cover
        item={{
          clickableSubtitle: true,
          id: 'some id',
          image: CoverReplacement,
          name: 'some name',
          onClick: () => console.log('Test'),
          subtitle: <span>'some subttle'</span>,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
