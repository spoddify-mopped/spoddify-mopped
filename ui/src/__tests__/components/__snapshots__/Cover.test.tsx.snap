// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover renders correctly with clickable subtitle 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  >
    <span>
      'some subttle'
    </span>
  </span>
</div>
`;

exports[`Cover renders correctly with custom class name 1`] = `
<div
  className="cover some class"
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`Cover renders correctly with image 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="some image url"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`Cover renders correctly with image and onClick 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`Cover renders correctly with required item properties 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`Cover renders correctly with subtitle 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  >
    <span>
      'some subttle'
    </span>
  </span>
</div>
`;
